### DISCUSSION TITLE
Help with API

### DISCUSSION THREAD

**Discussion by gboxxx** [Stirling-Tools/Stirling-PDF#268] [Q&A] [answered]
Hi, 

I am selfhosting on docker. I am able to send a request using HTTPie and merge two pdfs, however I'm not sure how to retrieve the resulting file. 

Outside of using an API client that lets me choose files directly, what's the proper way to pass files via API using something like n8n or any other platform that lets me write api calls but not necessarily allows me to "pick" files? I'm a bit confused by the documentation as it doesn't have much explaining. 

Thanks in advance!

**Comment by Frooodle** [comment, marked as answer]
Sorry i am being to improve my documentation as i move to a dedicated doc website....
As for your API issue most people just use a curl command
You can use it like talked about here https://github.com/Frooodle/Stirling-PDF/issues/131

So 
curl -X 'POST' \
  'http://192.168.2.43:8087/remove-password?password=actualpdfpassword' \
  -H 'accept: */*' \
  -H 'Content-Type: multipart/form-data' \
  -F 'fileInput=@02.PDF;type=application/pdf'
  
  or 
  
  curl --location '192.168.2.43:8087/remove-password' --form 'fileInput=@"/mypath/myfile.pdf"' --form 'password="pdf_password"' --output '/mypath/myfile_no_password.pdf'
  
  depending on what format you want to use, both work
